src/CMakeLists.txt

```diff
@@ -363,6 +363,18 @@ ADD_SUBDIRECTORY(irc)
 
 ADD_DEFINITIONS(-DAPP_VERSION="${APP_VERSION}" -DAPP_BUILD="${APP_BUILD}" -DAPP_TARGET="${APP_TARGET}")
 
+IF(UNIX)
+    SET(DATAROOTDIR "${CMAKE_INSTALL_PREFIX}/share" CACHE STRING "The platform-specific directory in which read-only data is generally installed")
+    SET(DATADIR "${DATAROOTDIR}/mudlet" CACHE STRING "Directory in which mudlet installs its read-only data")
+    SET(LUA_DEFAULT_DIR "${DATADIR}/lua" CACHE STRING "Directory in which mudlet installs its read-only lua scripts")
+ENDIF(UNIX)
+
+# Define a preprocessor symbol with the default fallback location from which
+# to load installed mudlet lua files. Set LUA_DEFAULT_DIR to a
+# platform-specific value. If LUA_DEFAULT_DIR is unset, the root directory
+# will be used.
+ADD_DEFINITIONS(-DLUA_DEFAULT_PATH="${LUA_DEFAULT_DIR}")
```

Comment:
> Ideally we also want to provide the input for the "sudo make install" stuff that CMake will brew up given the right conditions - because we are trying to ensure that the compiled in value matches what the user actually ends up with

What should be added beyond DATADIR and LUA_DEFAULT_DIR?

If you mean to add the INSTALL() cmake rules, how about making that a separate issue? CMake doesn't create a "make install" target at all atm. CMake doesn't work at all for installation. That's entirely missing. Only qmake creates a "make install" target.